### DISCUSSION TITLE
One-to-Many relationship is not being reflected in types.

### DISCUSSION THREAD

**Discussion by jearthman** [prisma/prisma#20730] [Q&A] [answered]
### Question

I'm having an issue using a one-to-many object in my app. I have a Chat model that has many Messages. schema snippet:

```prisma
model Chat {
  id                Int                @id @default(autoincrement())
  createdAt         DateTime           @default(now()) @db.Timestamp(6)
  messages          Message[]
  studentAssignment StudentAssignment? @relation()
}

enum MessageRole {
  system
  user
  assistent
}

model Message {
  id        Int         @id @default(autoincrement())
  role      MessageRole
  content   String      @db.Text
  timestamp DateTime    @default(now()) @db.Timestamp(6)
  chatId    Int
  chat      Chat        @relation(fields: [chatId], references: [id])
}
```

I'm getting the chat with this snippet:

```typescript
const chat = await prisma.chat.findUnique({
      where: {
        id: parseInt(chatId),
      },
      include: {
        messages: true,
      },
    });
```
    
 In the component where I'm having the issue I'm importing Chat from .prisma/client but it's missing 'messages' from it's type.
 
 ```typescript
type Chat = {
    id: number;
    createdAt: Date;
}
```

So when I try to use chat.messages TS complains that messages doesn't exist on Chat. 

I've migrated and regenerated the .prisma/client folder but get the same result. The foreign key constraint for Messages is being set up correctly in the DB. What's the correct way to fix this without assertion?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment, marked as answer]
Hello @jearthman 👋 

By default, only the scalar fields are returned in the generated types, due to which `messages` is not included in the `Chat` type.
You can use `Prisma.validator` to get the return type which included the `messages` field. ([Reference](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type))

It would look something like this:
```ts
import { Prisma } from '@prisma/client'

// 1: Define a type that includes the relation to `Post`
const chatWithMessages = Prisma.validator<Prisma.ChatDefaultArgs>()({
  include: { messages: true },
})

type chatWithMessages = Prisma.ChatGetPayload<typeof chatWithMessages>
```

You can then use `chatWithMessages` type in the `findUnique` query.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
